mismi-s3/src/Mismi/S3/Data.hs

```diff
@@ -74,6 +74,8 @@ data S3Error =
   | DestinationAlreadyExists Address
   | DestinationDoesNotExist Address
   | DestinationFileExists FilePath
+  | DestinationNotDirectory FilePath
+  | DestinationMissing FilePath
```

Comment:
Would it make sense to have a different wrapper error type for `downloadRecursive`? /justaskin

Reply:
Maybe. What I have here is just the simplest thing that works. It will probably need to behave differently for lots of small files vs lots of big file.